docs/garbage-collection.md

```diff
@@ -0,0 +1,163 @@
+	if err != nil {
+		return err
+	}
+	defer done(ctx)
+```
+
+This will create a lease which will defer its own deletion and have a default
+expiry of 24 hours (in case the process dies before defer). For most use case,
+this is enough and no more thought needs to be put into leases.
+
+_But, of course, more complicated use cases are supported..._
+
+If the program or lease are intended to be longer lived, instead of the very
+easy `client.WithLease`, the lease manager can be used directly. This also
+allows for setting custom labels on the lease or manipulating its resources.
+Use `client.LeasesService()` to get a [lease Manager](https://godoc.org/github.com/containerd/containerd/leases#Manager)
+which can be used to create, list, and delete leases as well as manage the
+referenced resources for that lease.
+
+```.go
+	manager := client.LeasesService()
+
+	// this lease will never expire
+	// Use `leases.WithExpiration` to make it expire
+	// Use `leases.WithLabels` to apply any labels
+	l, err := manager.Create(ctx, leases.WithRandomID())
+	if err != nil {
+		return err
+	}
+
+	// Update current context to add lease
+	ctx = leases.WithLease(ctx, l.ID)
+
+	// Do something, lease will be used...
+
+	// Delete lease at any time, or track it to delete later
+	if err := ls.Delete(ctx, l); err != nil {
+		return err
+	}
+```
+
+
+### using gRPC
+
+The lease is not an explicit field in the API (except of course the leases
+service), but rather an optional field any API service can use. Leases can
+be set on any gRPC service endpoint using a gRPC header. Set the
+gRPC header `containerd-lease` to the lease identifier and the API
+service will operate within that lease context.
+
+To manage the creation and deletion of leases, use the leases gRPC service.
+
+## Garbage Collection labels
+
+The garbage collection defines relationships between resources in two different
+ways, by type specific resources properties, and by resource labels. The type
+specific properties do not need to be managed by the user as they are part of
+the natural structure of a resource (i.e. a container's snapshot, a snapshot's
+parent, an image's target, etc). However resources may have relationships which
+are not defined by `containerd`, but rather by the client. For example, an OCI
+image has a manifest which references a config file and layer tars. These
+resources are stored in `containerd` as generic blobs, it is the client that
+understands the relationships between these blobs and sets them up using labels
+on the content resources.
+
+Resource labels can also be used to cue the garbage collector on other
+properties, such as expiry, whether an object should be kept without any
+reference or limit what is referenced.
+
+The supported garbage collection labels are...
+
+| Label key | Label value | Supported Resources | Description |
+|---|---|---|---|
+| `containerd.io/gc.root` | _nonempty_ | Content, Snapshots | Keep this object and anything it references |
```

Comment:
I _think_ the convention was to set RFC3339Nano timestamp?

Reply:
Added a mention in the description. The value is unread but the client uses the timestamp since if nothing else, it is useful to know when it was added.